src/dispatch-payload.ts

```diff
@@ -61,47 +70,21 @@ export const DispatchPayload = async (
   let resp: any = {};
 
   try {
-    switch (eventType) {
-      case EventTypes.FUNCTION_EXECUTED:
-        resp = await RunFunction(
-          payload as InvocationPayload<FunctionInvocationBody>,
-          functionModule,
-        );
-        break;
-      case EventTypes.BLOCK_ACTIONS:
-        resp = await RunBlockAction(
-          payload as InvocationPayload<BlockActionInvocationBody>,
-          functionModule,
-        );
-        break;
-      case EventTypes.BLOCK_SUGGESTION:
-        resp = await RunBlockSuggestion(
-          payload as InvocationPayload<BlockSuggestionInvocationBody>,
-          functionModule,
-        );
-        break;
-      case EventTypes.VIEW_SUBMISSION:
-        resp = await RunViewSubmission(
-          payload as InvocationPayload<ViewSubmissionInvocationBody>,
-          functionModule,
-        );
-        break;
-      case EventTypes.VIEW_CLOSED:
-        resp = await RunViewClosed(
-          payload as InvocationPayload<ViewClosedInvocationBody>,
-          functionModule,
-        );
-        break;
-      default:
-        throw new UnhandledEventError(
-          `Received an unsupported event of type: "${eventType}" for the ${functionCallbackId} function.`,
-        );
+    if (!EVENT_TO_HANDLER_MAP[eventType]) {
+      throw new UnhandledEventError(
+        `Received an unsupported event of type: "${eventType}" for the ${functionCallbackId} function.`,
+      );
+    } else {
+      resp = await EVENT_TO_HANDLER_MAP[eventType](
+        baseHandlerArgs,
+        functionModule,
+      );
```

Comment:
NIT: code style preference, I don't think this explicit `else` is required, I would tend towards removing it to reduce indentation but not sure if the rest of this file follows this pattern
```suggestion
    }
    resp = await EVENT_TO_HANDLER_MAP[eventType](
      baseHandlerArgs,
      functionModule,
    );
```